src/_pytest/capture.py

```diff
@@ -136,41 +136,59 @@ def suspend_global_capture(self, in_=False):
 
     def suspend(self, in_=False):
         # Need to undo local capsys-et-al if it exists before disabling global capture.
-        self.suspend_fixture(self._current_item)
+        self.suspend_fixture()
         self.suspend_global_capture(in_)
 
     def resume(self):
         self.resume_global_capture()
-        self.resume_fixture(self._current_item)
+        self.resume_fixture()
 
     def read_global_capture(self):
         return self._global_capturing.readouterr()
 
     # Fixture Control (it's just forwarding, think about removing this later)
 
-    def activate_fixture(self, item):
+    @contextlib.contextmanager
+    def _item_fixture(
+        self, request: FixtureRequest
+    ) -> Generator["CaptureFixture", None, None]:
+        if self._item_capture:
+            other_name = next(
+                k
+                for k, v in map_fixname_class.items()
+                if v is self._item_capture.captureclass
+            )
+            raise request.raiseerror(
+                "cannot use {} and {} at the same time".format(
+                    request.fixturename, other_name
+                )
+            )
+        capture_class = map_fixname_class[request.fixturename]
+        self._item_capture = CaptureFixture(capture_class, request)
```

Comment:
TBH I prefer the old name (`_capture_fixture`), because it matches the type name.